### ISSUE TITLE
How to get score for all classes

### ISSUE THREAD

**Issue by dangmanhtruong1995** [qqwweee/keras-yolo3#592]
Hi, how do I get scores for all classes? For example, there is a dog object, with score 0.9. However I also want the cat score for the same object (0.1). How do I get them ? Thank you very much.

**Comment by datduonguva** [comment]
The model uses logistic output, not a softmax. So if dog score is 0.9, cat
score could be any value from 0.0 to 1.0, not just 0.1.
If you still want it, the model output has shape N*N*(5+ n_class). The
columns [5:] in the last dimension contain those values.

**Comment by dangmanhtruong1995** [comment]
Hi. Thanks for the quick reply. I noticed that in yolo.py file, there is this part: 
```
out_boxes, out_scores, out_classes = self.sess.run(
            [self.boxes, self.scores, self.classes],
            feed_dict={
                self.yolo_model.input: image_data,
                self.input_image_shape: [image.size[1], image.size[0]],
                K.learning_phase(): 0
            })
```

However when I checked further, it did not output scores for all classes. So how do I get columns [5:] ? Thank you very much.

**Comment by datduonguva** [comment]
in yolo3/model.py, change the yolo_eval to: 
return boxes_, scores_, classes_, yolo_outputs[0], yolo_outputs[1], yolo_outputs[2]

in yolo.py

in line 45: 
self.boxes, self.scores, self.classes, self.output_1, self.output_2, self.output_3= self.generate()
in line 97: 
boxes, scores, classes, output_1, output_2, output_3 = yolo_eval(self.yolo_model.output, self.anchors,


in line 100: 
return boxes, scores, classes, output_1, output_2, output_3

in line 119:
out_boxes, out_scores, out_classes, out1, out2, out_3 = self.sess.run(

the out1, out2, out3 should have shape batchx13x13x(5+n_class), batchx26x26x(5+n_class), batchx52x52x(5+n_class)
the class score will then be out1[:, :, :, 5:], out2[:, :, :, 5:], out2[:, :, :, 5:]

**Comment by dangmanhtruong1995** [comment]
Hi, can you explain to me what out1, out2, out3 means? Because I thought there would only be 1 more variable to store the class scores. Thank you very much.

**Comment by datduonguva** [comment]
The darknet body creates 3 outputs, not 1, whose shapes are batchx13x13x(5+n_class), batchx26x26x(5+n_class), batchx52x52x(5+n_class). Yolo_eval  function uses an algorithms called non_max_suppression (you can find in the code) to select which output to use.

**Comment by dangmanhtruong1995** [comment]
Hi, I'm sorry but I'm really new to tensorflow, so it is very difficult for me to look into the code and modify stuff for my need, so I wonder how can I get the class scores for each bounding box using non_max_suppression ? If you could help me that would be great, because this looks really really complicated :)
